src/main/java/io/javalin/http/ResponseHandler.kt

```diff
@@ -0,0 +1,69 @@
+package io.javalin.http
+
+import io.javalin.core.compression.CompressionStrategy
+import io.javalin.core.util.Header
+import javax.servlet.ServletOutputStream
+import javax.servlet.WriteListener
+import javax.servlet.http.HttpServletResponse
+import javax.servlet.http.HttpServletResponseWrapper
+
+/**
+ * This class acts as an "interceptor" for handling both static and dynamic responses.
+ * We use it to wrap the "res" parameter inside JavalinServlet#service
+ *
+ * For dynamic responses, we then simply call res.outputStream.write() inside service#writeResult and feed it the resultStream
+ *
+ * For static responses, we intercept Jetty's writing by feeding its ResourceHandler the wrapped "res" param instead of the original.
+ * Jetty uses the same outputStream.write() method to write static content to output after handling it.
+ * Because of this, we are able to capture Jetty's output in here, modify it as needed, then commit it ourselves
+ *
+ * For the moment, this is being used to bypass Jetty's compression handling, so we can use our own (which supports Brotli and dynamic content),
+ * however, there could be more uses for this in the future.
+ */
+class ResponseHandler(
+        response: HttpServletResponse,
+        ctx: Context,
+        compressionStrategy: CompressionStrategy) : HttpServletResponseWrapper(response)
+{
+    private val outputWrapper = OutputWrapper(response, ctx, compressionStrategy)
+    override fun getOutputStream() = outputWrapper
+
+    class OutputWrapper(
+            private val res: HttpServletResponse,
+            private val ctx: Context,
+            private val compressionStrategy: CompressionStrategy) : ServletOutputStream()
+    {
+        override fun isReady(): Boolean = res.outputStream.isReady
+        override fun setWriteListener(writeListener: WriteListener?) = res.outputStream.setWriteListener(writeListener)
+        override fun write(b: Int) {
```

Comment:
This needs a comment, I guess. Shouldn't it also call `super.write` ?

Reply:
The result should the same, since we're retrieving the unmodified outputStream and calling its write method. Haven't tested yet though

Reply:
Ah, right. Just tried it and remembered why I didn't put super.write()

It's because this class is extending ServletOutputStream, which is an abstract class that doesn't implement the write(int) method. 

So, by calling res.outputStream.write(int), we are effectively calling super.write(int), as I explained in my previous comment.